### DISCUSSION TITLE
Consider not modifying environment variables on startup by default

### DISCUSSION THREAD

**Discussion by dsherret** [PowerShell/PowerShell#24347] [Enhancements]
I just wanted to raise that powershell injecting `C:\Program Files\PowerShell\7;` to the PATH env var on startup causes some issues.

One scenario I've run into: I have an application that does a full rebuild when the PATH changes instead of an incremental build. This is due to a dependency and isn't something I have control over. My editor builds the application (in this case, using rust analyzer) and that has a specific PATH, but then when I go to build in powershell (ex. `cargo build`), powershell has a different PATH which causes a full rebuild because the PATH has changed since the last time I've built. It's quite annoying. Anyway, just wanted to recommend reconsidering this and to instead do it on installation (or don't do it at all... I can't see anything really useful in `C:\Program Files\PowerShell\7` that's executable).

(The workaround for this is to launch the editor from powershell first)

**Comment by rhubarb-geek-nz** [comment]
Hi, I am very sympathetic, PATH is one of the most important environment variables and in my opinion shells should not change it.

I would have thought that you want to change it, then do so in your profile.

This was looked at here https://github.com/PowerShell/PowerShell/issues/24090

PowerShell already has PSModulePath for finding things, I don't see why it needs to hijack PATH.
